### PR TITLE
Handle continue as new in resend

### PR DESCRIPTION
**Pull request by yux0**

### CODE REVIEW

File: common/xdc/nDCHistoryResender.go [hunk 1]
```diff
@@ -128,13 +128,35 @@ func (n *NDCHistoryResenderImpl) SendSingleWorkflowHistory(
...
+			resp, err := n.getHistory(
+				domainID,
+				workflowID,
+				newRunID,
+				common.Int64Ptr(common.FirstEventID-1),
+				common.Int64Ptr(lastEvent.GetVersion()),
+				nil,
+				nil,
+				nil,
+				defaultPageSize)
```
Comment (wxing1292) [inline common/xdc/nDCHistoryResender.go]: set this to 1?

File: common/xdc/nDCHistoryResender.go [hunk 2]
```diff
@@ -261,3 +282,34 @@ func (n *NDCHistoryResenderImpl) getHistory(
...
+}
+
+func (n *NDCHistoryResenderImpl) deserializeBlob(blob *shared.DataBlob) ([]*shared.HistoryEvent, error) {
+
+	var err error
+	var historyEvents []*shared.HistoryEvent
+
+	switch blob.GetEncodingType() {
+	case shared.EncodingTypeThriftRW:
+		historyEvents, err = n.serializer.DeserializeBatchEvents(&persistence.DataBlob{
```
Comment (wxing1292) [inline common/xdc/nDCHistoryResender.go]: just return here?

File: common/xdc/nDCHistoryResender.go [hunk 3]
```diff
@@ -128,13 +128,35 @@ func (n *NDCHistoryResenderImpl) SendSingleWorkflowHistory(
...
 		// we don't need to handle continue as new here
 		// because we only care about the current run
-		// TODO: revisit to evaluate if we need to handle continue as new
 		replicationRequest := n.createReplicationRawRequest(
 			domainID,
 			workflowID,
 			runID,
 			historyBatch.rawEventBatch,
 			historyBatch.versionHistory.GetItems())
+		lastEvent, err := n.getLastEvent(historyBatch.rawEventBatch)
```
Comment (wxing1292) [inline common/xdc/nDCHistoryResender.go]: here is trying to decode every history event batch for the last event in that batch, try only decode the last batch

File: common/xdc/nDCHistoryResender.go (hunk 1 above)
Comment (yux0) [reply common/xdc/nDCHistoryResender.go]: done

File: common/xdc/nDCHistoryResender.go (hunk 2 above)
Comment (yux0) [reply common/xdc/nDCHistoryResender.go]: updated

File: common/xdc/nDCHistoryResender.go (hunk 3 above)
Comment (yux0) [reply common/xdc/nDCHistoryResender.go]: good suggestion. done

File: common/xdc/nDCHistoryResender.go [hunk 4]
```diff
@@ -126,15 +127,38 @@ func (n *NDCHistoryResenderImpl) SendSingleWorkflowHistory(
...
+					common.Int64Ptr(lastEvent.GetVersion()),
+					nil,
+					nil,
+					nil,
+					continueAsNewPageSize)
+				if err != nil {
+					return err
+				}
+				replicationRequest.NewRunEvents = resp.HistoryBatches[0]
+			}
```
Comment (wxing1292) [inline common/xdc/nDCHistoryResender.go]: how about turning this into a function?

Comment (wxing1292) [reply common/xdc/nDCHistoryResender.go]: plz also double check on the slice size `replicationRequest.NewRunEvents = resp.HistoryBatches[0]`

File: service/history/nDCActivityReplicator_test.go [hunk 5]
```diff
@@ -672,7 +672,7 @@ func (s *activityReplicatorSuite) TestSyncActivity_VersionHistories_SameSchedule
 	}
 	s.mockMutableState.EXPECT().GetVersionHistories().Return(localVersionHistories).AnyTimes()
 	s.mockMutableState.EXPECT().GetActivityInfo(scheduleID).Return(nil, false).AnyTimes()
-	s.mockMutableState.EXPECT().IsWorkflowExecutionRunning().Return(true).AnyTimes()
+	s.mockMutableState.EXPECT().GetWorkflowStateCloseStatus().Return(0, 0).AnyTimes()
```
Comment (wxing1292) [inline service/history/nDCActivityReplicator_test.go]: `persistence.WorkflowStateCreated, persistence.WorkflowCloseStatusNone`

File: service/history/nDCActivityReplicator_test.go [hunk 6]
```diff
@@ -748,7 +748,7 @@ func (s *activityReplicatorSuite) TestSyncActivity_VersionHistories_LocalVersion
...
```
Comment (wxing1292) [inline service/history/nDCActivityReplicator_test.go]: `persistence.WorkflowStateCreated, persistence.WorkflowCloseStatusNone`

**Pull request merged**